### PR TITLE
steps: bump requirements / fix build on Apple M1

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

File: bluebrain/repo-patches/packages/steps/package.py [hunk 1]
```diff
@@ -88,9 +88,10 @@ class Steps(CMakePackage):
     depends_on("py-scipy", type=("build", "test", "run"))
     depends_on("python", type=("build", "test", "run"))
     depends_on("random123", when="~bundle")
-    depends_on("sundials@:2.99.99+int64", when="~bundle")
+    depends_on("sundials@:2.99.99+int64", when="@:5~bundle")
+    depends_on("sundials@:6.99+int64", when="@develop~bundle")
```
Comment (matz-e) [inline bluebrain/repo-patches/packages/steps/package.py]: Is the `.99` a range? Spack now accepts `@:2` to mean everything up to, excluding `3`

File: var/spack/repos/builtin/packages/petsc/package.py
Comment (matz-e) [inline var/spack/repos/builtin/packages/petsc/package.py:1]: Is this going to get upstreamed?

File: bluebrain/repo-patches/packages/steps/package.py (hunk 1 above)
Comment (tristan0x) [reply bluebrain/repo-patches/packages/steps/package.py]: thank you for pointing that out. It fixed in f1545503c5

File: var/spack/repos/builtin/packages/petsc/package.py
Comment (tristan0x) [reply var/spack/repos/builtin/packages/petsc/package.py:1]: I will try to upstream that to PETSc.

**Review by matz-e — APPROVED** [review]

**Pull request merged**